### PR TITLE
add remote-mode to startWorker e2e tests

### PR DESCRIPTION
**Pull request by RamIdeas**
## What this PR solves / how to test

This is a follow-up to https://github.com/cloudflare/workers-sdk/pull/6124 and enables the same tests to be run in remote mode as well as local mode.

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [x] Required / Maybe required
  - [ ] Not required because:
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [ ] Included
  - [x] Not necessary because:
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): 
  - [x] Not necessary because:

### CODE REVIEW

File: packages/wrangler/e2e/vitest.setup.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+declare module "vitest" {
+	interface Assertion extends CustomMatchers {}
+	interface AsymmetricMatchersContaining extends CustomMatchers {}
+}
+
+expect.extend({
+	// Extend vitest's expect object so that it has our new matcher
+	toContainMatchingObject(received: object[], expected: object) {
+		// const matcher = createMatcher(expected);
```
Comment (petebacondarwin) [inline packages/wrangler/e2e/vitest.setup.ts:15]: Remove?

File: packages/wrangler/src/api/startDevWorker/LocalRuntimeController.ts [hunk 2]
```diff
@@ -151,6 +151,7 @@ export class LocalRuntimeController extends RuntimeController {
 					await convertToConfigBundle(data),
 					this.#proxyToUserWorkerAuthenticationSecret
 				);
+			options.liveReload = false; // TODO: set in buildMiniflareOptions once old code path is removed
```
Comment (petebacondarwin) [inline packages/wrangler/src/api/startDevWorker/LocalRuntimeController.ts:154]: Issue to track this?

File: packages/wrangler/src/api/startDevWorker/ProxyController.ts [hunk 3]
```diff
@@ -161,6 +161,7 @@ export class ProxyController extends Controller<ProxyControllerEventMap> {
 					logger.loggerLevel === "debug" ? "wrangler-ProxyWorker" : "wrangler",
 			}),
 			handleRuntimeStdio,
+			liveReload: false,
```
Comment (petebacondarwin) [inline packages/wrangler/src/api/startDevWorker/ProxyController.ts:164]: Is the default true? If so, that was an interesting decision on the part of Miniflare...

**Review by petebacondarwin — APPROVED** [review]

File: packages/wrangler/src/api/startDevWorker/ProxyController.ts (hunk 3 above)
Comment (RamIdeas) [reply packages/wrangler/src/api/startDevWorker/ProxyController.ts:164]: Seems so. Only affects responses with html content type but yeah agree. startWorker's default is false

File: packages/wrangler/src/api/startDevWorker/LocalRuntimeController.ts (hunk 2 above)
Comment (RamIdeas) [reply packages/wrangler/src/api/startDevWorker/LocalRuntimeController.ts:154]: Don't think that's necessary. It's just a refactor. No behaviour difference or heavy uplift.

File: packages/wrangler/e2e/vitest.setup.ts (hunk 1 above)
Comment (RamIdeas) [reply packages/wrangler/e2e/vitest.setup.ts:15]: Argh. Will do it in another PR to avoid going through CI hell again

**Pull request merged**